src/jit/decomposelongs.cpp

```diff
@@ -796,11 +799,9 @@ GenTree* DecomposeLongs::DecomposeInd(LIR::Use& use)
         addrBase->TypeGet(), addrBase->AsLclVarCommon()->GetLclNum(), BAD_IL_OFFSET);
     GenTreePtr addrHigh = new(m_compiler, GT_LEA) GenTreeAddrMode(TYP_REF, addrBaseHigh, nullptr, 0, genTypeSize(TYP_INT));
     GenTreePtr indHigh = new (m_compiler, GT_IND) GenTreeIndir(GT_IND, TYP_INT, addrHigh, nullptr);
-    
+
     // Insert the nodes into the block
-    m_blockRange.InsertAfter(addrBaseHigh, indLow);
-    m_blockRange.InsertAfter(addrHigh, addrBaseHigh);
-    m_blockRange.InsertAfter(indHigh, addrHigh);
+    m_blockRange.InsertAfter(LIR::SeqTree(m_compiler, indHigh), indLow);
```

Comment:
I don't think that correct. There are two cases:
1. All operands have been replaced with local variables using `ReplaceWithLclVar`
2. Some/all operands have not been replaced with local variables

In the first case, the local var use that is created to replace the use of the spilled tree is threaded into the LIR block immediately after the store of the spilled tree. See the comment above `ReplaceWithLclVar` for example IR. In this case it should be sufficient to remove the local var uses from the block before calling `SeqTree` as these variables are guaranteed to be unaliased.

In the second case, the operands must be present in the LIR block by construction.  In this case, it is neither sufficient nor necessarily safe to remove the operands from the LIR block before calling `SeqTree`, as sequencing the tree and then inserting the new range may have moved the operands across nodes with conflicting side effects.

In both cases, I think it is simpler and safer to just insert the new IR manually.